File: backend/src/main/java/com/etekhno/rentacar/domain/UserDTO.java
package com.etekhno.rentacar.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
public class UserDTO {
    String id;
    String firstName;
    String lastName;
    String email;
    String token;
    String timezone;
    String locale;

    public UserDTO(String id, String firstName, String lastName, String email) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
    }
    public UserDTO(String id, String firstName, String lastName, String email, String timezone, String locale) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
        this.timezone = timezone;
        this.locale = locale;
    }
}

File: frontend/lib/domains/user_dto.dart
class UserDTO {
  String id;
  String firstName;
  String lastName;
  String email;
  String? token;
  String? timezone;
  String? locale;

  UserDTO({
    required this.id,
    required this.firstName,
    required this.lastName,
    required this.email,
    this.token,
    this.timezone,
    this.locale,
  });

  factory UserDTO.fromJson(Map<String, dynamic> json) {
    return UserDTO(
      id: json['id'] as String,
      firstName: json['firstName'] as String,
      lastName: json['lastName'] as String,
      email: json['email'] as String,
      token: json['token'] as String?,
      timezone: json['timezone'] as String?,
      locale: json['locale'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'firstName': firstName,
      'lastName': lastName,
      'email': email,
      'token': token,
      'timezone': timezone,
      'locale': locale,
    };
  }
}
